File: ruby/arcade/minesweeper.rb
def minesweeper(matrix)
	w, h = matrix[0].size, matrix.size
	board = Array.new(h, Array.new(w, 0))
	h.times { |i| w.times { |j| board[i][j] = 0 } }
	h.times do |r|
		w.times do |c|
			board[r][c] += (matrix[r - 1][c - 1]) ? 1 : 0
			board[r][c] += (matrix[r - 1][c]) ? 1 : 0
			board[r][c] += (matrix[r - 1][c + 1]) ? 1 : 0
			board[r][c] += (matrix[r][c - 1]) ? 1 : 0
			board[r][c] += (matrix[r][c + 1]) ? 1 : 0
			board[r][c] += (matrix[r + 1][c - 1]) ? 1 : 0
			board[r][c] += (matrix[r + 1][c]) ? 1 : 0
			board[r][c] += (matrix[r + 1][c + 1]) ? 1 : 0
		end
	end
	return board
end

File: Arcade/Intro/Java/Minesweeper.java
final class Minesweeper {

	int[][] minesweeper(boolean[][] matrix) {
	    int w = matrix[0].length, h = matrix.length;
	    int[][] board = new int[h][w];
	    
	    for(int i = 0; i < h; i++) {
	        for(int j = 0; j < w; j++) {
	            if(matrix[i][j]) {
	                if(i + 1 < h) board[i + 1][j]++;
	                if(i + 1 < h && j - 1 >= 0) board[i + 1][j - 1]++;
	                if(j - 1 >= 0) board[i][j - 1]++;
	                if(i - 1 >= 0 && j + 1 < w) board[i - 1][j + 1]++;
	                if(i - 1 >= 0) board[i - 1][j]++;
	                if(i - 1 >= 0 && j - 1 >= 0) board[i - 1][j - 1]++;
	                if(j + 1 < w) board[i][j + 1]++;
	                if(i + 1 < h && j + 1 < w) board[i + 1][j + 1]++;
	            }
	        }
	    }
	    
	    return board;
	}
}